File: desktop/src/main/java/com/corrodinggames/rts/gameFramework/graphics/TeamColorTexture.java
package com.corrodinggames.rts.gameFramework.graphics;

import android.graphics.Bitmap;
import com.corrodinggames.rts.game.ColorMode;
import com.corrodinggames.rts.game.PlayerTeam;
import com.corrodinggames.rts.gameFramework.AssetType;
import com.corrodinggames.rts.gameFramework.GameEngine;
import com.corrodinggames.rts.gameFramework.PerformanceProfiler;
import java.io.IOException;

/* JADX INFO: renamed from: com.corrodinggames.rts.gameFramework.m.h */
/* JADX INFO: loaded from: game-lib.jar:com/corrodinggames/rts/gameFramework/m/h.class */
public class TeamColorTexture extends Texture {
    public static ShaderProgram x;
    public static ShaderProgram y;
    public static ShaderProgram z;
    public static boolean A;
    boolean B = false;
    boolean C = false;
    private Texture H;
    private Texture I;
    int D;
    int E;
    ColorMode F;
    public static float G;

    public static synchronized void C() {
        if (A) {
            return;
        }
        try {
            GameEngine.log("Loading team shaders...");
            x = new TeamColorShader("assets/shaders/pureGreenTeamColor.frag", true);
            x.a("teamColor", -1);
            x.c();
            y = new TeamColorShader("assets/shaders/hueAddTeamColor.frag", false);
            y.a("teamColorAmount", 0.15f);
            y.a("teamColor", -1);
            y.c();
            z = new TeamColorShader("assets/shaders/hueShiftTeamColor.frag", false);
            z.a("teamColor", -1);
            z.c();
            A = true;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public void D() {
        if (!A) {
            C();
        }
    }

    @Override // com.corrodinggames.rts.gameFramework.graphics.Texture
    public String a() {
        if (this.H == null) {
            return "LazyColoring (error sourceBitmap==null)";
        }
        return "LazyColoring(" + this.E + "):" + this.H.a();
    }

    public TeamColorTexture(Texture texture, int i, ColorMode colorMode, int i2) {
        if (texture == null) {
            throw new RuntimeException("baseImage==null");
        }
        this.H = texture;
        this.D = i;
        this.F = colorMode;
        this.E = i2;
        this.H.a(this);
        this.k = null;
    }

    public void c(boolean z2) {
        if (GameEngine.areShadersSupported()) {
            if (z2) {
            }
            D();
            if (this.F == ColorMode.hueAdd) {
                a(y);
            } else if (this.F == ColorMode.hueShift) {
                a(z);
            } else {
                a(x);
            }
            this.I = this.H;
            this.C = true;
            return;
        }
        if (this.H.A()) {
            GameEngine.log("Lazy loaded bitmap using errored image: " + this.H.a());
            this.I = this.H;
            return;
        }
        if (z2) {
            try {
                GameEngine.log("Loading in lazy loaded bitmap:" + this.H.a() + " team:" + this.E);
            } catch (OutOfMemoryError e) {
                GameEngine.log("Colouring failed with OOM");
                GameEngine.reportOOM(AssetType.gameImageColor, e);
                this.I = GameEngine.getInstance().graphicsEngine2.r();
                return;
            }
        }
        long jA = PerformanceProfiler.a();
        this.H.i();
        this.I = this.H.clone();
        this.I.j();
        Texture[] textureArr = {this.I};
        int[] iArr = {this.D};
        int[] iArr2 = {this.E};
        long jA2 = PerformanceProfiler.a();
        if (this.F == ColorMode.hueAdd) {
            PlayerTeam.b(this.H, textureArr, iArr);
        } else if (this.F == ColorMode.hueShift) {
            PlayerTeam.a(this.H, textureArr, iArr, iArr2);
        } else {
            PlayerTeam.a(this.H, textureArr, iArr);
        }
        double dA = PerformanceProfiler.a(jA2);
        this.I.p();
        this.I.s();
        this.H.q();
        this.H = null;
        double dA2 = PerformanceProfiler.a(jA);
        if (dA2 > 1.0d) {
            GameEngine.log((this.F == ColorMode.pureGreen ? "Standard " : "Hue ") + "Colouring took:" + PerformanceProfiler.a(dA2) + " (" + PerformanceProfiler.a(dA) + ")");
        }
        G = (float) (((double) G) + dA2);
    }

    @Override // com.corrodinggames.rts.gameFramework.graphics.Texture
    public Bitmap b() {
        if (this.C && !GameEngine.areShadersSupported()) {
            GameEngine.log("Team shader coloring now disabled. Recoloring image: " + this.H.a());
            this.B = false;
            this.C = false;
            a((ShaderProgram) null);
        }
        if (!this.B) {
            c(true);
            this.B = true;
        }
        return this.I.k;
    }

    @Override // com.corrodinggames.rts.gameFramework.graphics.Texture
    public Texture c() {
        if (this.C && !GameEngine.areShadersSupported()) {
            GameEngine.log("Team shader coloring now disabled. Recoloring image: " + this.H.a());
            this.B = false;
            this.C = false;
            a((ShaderProgram) null);
        }
        if (!this.B) {
            if (G > 60.0f) {
            }
            c(true);
            this.B = true;
        }
        if (this.I == null) {
            throw new RuntimeException("coloredBitmap==null");
        }
        return this.I;
    }

    @Override // com.corrodinggames.rts.gameFramework.graphics.Texture
    public void w() {
        if (!this.B) {
            c(false);
            this.B = true;
        }
    }

    @Override // com.corrodinggames.rts.gameFramework.graphics.Texture
    public int u() {
        if ((!this.B) & (this.H != null)) {
            return this.H.u();
        }
        return super.u();
    }
}

File: core/src/main/kotlin/com/corrodinggames/rts/gameFramework/graphics/TeamColorTexture.kt
package com.corrodinggames.rts.gameFramework.graphics

import com.corrodinggames.rts.game.ColorMode
import com.corrodinggames.rts.gameFramework.GameEngine
import java.io.IOException

class TeamColorTexture(
    private val source: Texture?,
    val teamColor: Int,
    val colorMode: ColorMode,
    private val teamIndex: Int,
) : Texture() {
    val teamColorAmount: Float =
        if (colorMode == ColorMode.hueAdd) HUE_ADD_TEAM_COLOR_AMOUNT else 1f

    init {
        if (source == null) {
            throw RuntimeException("baseImage==null")
        }
        source.copyTextureSettingsTo(this)
        D()
    }

    override fun a(): String =
        if (source == null) {
            "LazyColoring (error sourceTexture==null)"
        } else {
            "LazyColoring($teamIndex):" + source.sourceName()
        }

    fun c(z2: Boolean) = Unit

    fun sourceTexture(): Texture =
        source ?: throw RuntimeException("team color source texture is null")

    fun resolveSourceTexture(): Texture = sourceTexture()

    override fun b(): IntArray =
        resolveSourceTexture().editablePixels() ?: IntArray(0)

    override fun c(): Texture = resolveSourceTexture()

    override fun w() = Unit

    override fun u(): Int =
        source?.estimatedMemoryBytes() ?: super.estimatedMemoryBytes()

    fun D() {
        ensureTeamShaders()
        a(
            when (colorMode) {
                ColorMode.hueAdd -> hueAddShader
                ColorMode.hueShift -> hueShiftShader
                else -> pureGreenShader
            }
        )
    }

    companion object {
        const val HUE_ADD_TEAM_COLOR_AMOUNT = 0.15f

        @JvmStatic
        var pureGreenShader: ShaderProgram? = null
            private set

        @JvmStatic
        var hueAddShader: ShaderProgram? = null
            private set

        @JvmStatic
        var hueShiftShader: ShaderProgram? = null
            private set

        @JvmStatic
        var teamShadersLoaded: Boolean = false
            private set

        @JvmStatic
        @Synchronized
        fun C() {
            if (teamShadersLoaded) {
                return
            }
            try {
                GameEngine.log("Loading team shaders...")
                pureGreenShader = TeamColorShader("assets/shaders/pureGreenTeamColor.frag", true).also {
                    it.a("teamColor", -1)
                    it.c()
                }
                hueAddShader = TeamColorShader("assets/shaders/hueAddTeamColor.frag", false).also {
                    it.a("teamColorAmount", HUE_ADD_TEAM_COLOR_AMOUNT)
                    it.a("teamColor", -1)
                    it.c()
                }
                hueShiftShader = TeamColorShader("assets/shaders/hueShiftTeamColor.frag", false).also {
                    it.a("teamColor", -1)
                    it.c()
                }
                teamShadersLoaded = true
            } catch (e: IOException) {
                throw RuntimeException(e)
            }
        }

        private fun ensureTeamShaders() = C()
    }
}
